### PR TITLE
Create subscription of plan

### PR DESCRIPTION
**Pull request by GabrielDeveloper**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| **Issue**         | https://mundipagg.atlassian.net/browse/MOD-900
| **What?**         | Create subscription of the plan
| **Why?**          | To be able to subscription a plan
| **How?**          | Get saved plans information and create subscriptions at Mundipagg

### CODE REVIEW

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage decreased per file
===========================
- src/Recurrence/Aggregates/Plan.php  -2
- src/Recurrence/Services/RecurrenceService.php  -21
- src/Recurrence/Aggregates/Subscription.php  -1
         

Complexity increasing per file
==============================
- tests/Recurrence/ValueObjects/PlanIdTest.php  1
- src/Recurrence/Services/RecurrenceService.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5105251&bid=16789724)

File: src/Recurrence/Repositories/PlanRepository.php [hunk 1]
```diff
@@ -161,7 +145,20 @@ public function delete(AbstractEntity $object)
     }
     public function findByMundipaggId(AbstractValidString $mundipaggId)
     {
+        $table = $this->db->getTable(
+            AbstractDatabaseDecorator::TABLE_RECURRENCE_PRODUCTS_PLAN
+        );
+        $objectId = $mundipaggId->getValue();
+
+        $query = "SELECT * FROM $table WHERE plan_id = '$objectId' LIMIT 1";
```
Comment (wallaceSF) [inline src/Recurrence/Repositories/PlanRepository.php]: adicione chaves na variável

File: src/Recurrence/Services/SubscriptionService.php [hunk 2]
```diff
@@ -252,16 +278,34 @@ private function fillBoletoData(&$subscription)
...
-    private function fillInterval(&$subscription)
+    private function fillPlanId($subscription, $plan)
     {
+        if ($plan !== null) {
+            $subscription->setPlanId($plan->getMundipaggId());
+        }
+        return null;
+    }
+
+    private function fillInterval(&$subscription, Plan $plan = null)
```
Comment (wallaceSF) [inline src/Recurrence/Services/SubscriptionService.php]: Objeto é sempre referência

File: src/Recurrence/Services/SubscriptionService.php [hunk 3]
```diff
@@ -252,16 +278,34 @@ private function fillBoletoData(&$subscription)
         $subscription->setBoletoDays($boletoDays);
     }
 
-    private function fillSubscriptionItems(&$subscription, $order)
+    private function fillSubscriptionItems(&$subscription, $order,Plan $plan = null)
```
Comment (michelpl) [inline src/Recurrence/Services/SubscriptionService.php]: Adicionar espaço depois da última vírgula

Comment (michelpl) [reply src/Recurrence/Services/SubscriptionService.php]: Faz sentido enviar o plano só pra dar return? Isso não deveria ser verificado antes de chamar a funcão?

File: src/Recurrence/Services/SubscriptionService.php [hunk 4]
```diff
@@ -283,8 +327,12 @@ private function fillInterval(&$subscription)
         $subscription->setIntervalCount($intervalCount);
     }
 
-    private function fillDescription(&$subscription)
+    private function fillDescription(&$subscription, Plan $plan = null)
     {
+        if ($plan !== null) {
```
Comment (michelpl) [inline src/Recurrence/Services/SubscriptionService.php]: Mesma coisa do de cima

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- tests/Recurrence/ValueObjects/PlanIdTest.php  1
- src/Recurrence/Services/RecurrenceService.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5105251&bid=16789724)

**Review by michelpl — APPROVED** [review]

**Review by wallaceSF — APPROVED** [review]

**Pull request merged**